superset/models/core.py

```diff
@@ -85,6 +85,37 @@ def copy_dashboard(mapper, connection, target):
     session.commit()
 
 
+def query_obj_backfill(form_data, datasource_id=None, datasource_type=None):
```

Comment:
There was a time where `form_data` did not include datasource_id & datasource_type, but now it always does in the shape of `fd.get('datasource') == '231__sqla'` or something like that.

Reply:
Good to know! I'll remove them, which should clean up the signature of this method.